decorating/animation.py

```diff
@@ -87,27 +87,35 @@ def _spinner(control):
 
 
 # deal with it
-def animated(func):
-    @wraps(func)
-    def wrapper(*args, **kwargs):
-        global last_thread
-        if not wrapper.running:
-            sig.message = func.__name__
-            spinner_thread = threading.Thread(target=_spinner, args=(sig,))
-            spinner_thread.start()
-            last_thread = spinner_thread
-            wrapper.running = True
-        result = func(*args, **kwargs)
-        if wrapper.running:
-            sig.done = True
-            spinner_thread.join()
-            wrapper.running = False
-            sig.done = False
-
-        return result
-
-    wrapper.running = False
-
-    return wrapper
+def animated(func_or_message):
+    def _animated(func)
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            global last_thread
+            if not wrapper.running:
+                sig.message = name
+                spinner_thread = threading.Thread(target=_spinner, args=(sig,))
+                spinner_thread.start()
+                last_thread = spinner_thread
+                wrapper.running = True
+            result = func(*args, **kwargs)
+            if wrapper.running:
+                sig.done = True
+                spinner_thread.join()
+                wrapper.running = False
+                sig.done = False
+    
+            return result
+    
+        wrapper.running = False
+    
+        return wrapper
+        
+    if callable(func_or_message):  # function
+        name = func_or_message.__name__
+        return _animated(func_or_message)
```

Comment:
```
@animated('printer')
def slow():
     sleep(3)
     print('bar')

slow()
```
So this will works as expect, right? I was a little confuse because in your first example you put `awesome`, so I thinking could be pass a function too (however will be really odd)